pkg/collector/corechecks/snmp/internal/devicecheck/devicecheck.go

```diff
@@ -150,6 +155,21 @@ func (d *DeviceCheck) Run(collectionTime time.Time) error {
 		d.sender.ReportMetrics(d.config.Metrics, values, tags)
 	}
 
+	// Get a system appropriate ping check
+	if d.config.PingEnabled {
+		log.Infof("ATTEMPTING TO RUN PING FOR HOST: %s, tags: %+v", d.config.IPAddress, tags)
+		pingResult, err := d.Ping(d.config.PingConfig)
+		if err != nil {
+			log.Errorf("%s: failed to ping device: %s", d.config.IPAddress, err.Error())
+			d.sender.Gauge(pingCanConnectMetric, float64(0.0), tags)
+		} else {
+			log.Infof("%s: ping returned: %+v", d.config.IPAddress, pingResult)
+			d.sender.Gauge(pingAvgRttMetric, float64(pingResult.AvgRtt/time.Millisecond), tags)
+			d.sender.Gauge(pingCanConnectMetric, common.BoolToFloat64(pingResult.CanConnect), tags)
+			d.sender.Gauge(pingPacketLoss, pingResult.PacketLoss, tags)
+		}
+	}
```

Comment:
Can we just use a go routine here to avoid blocking? If we do so, few notes:
1/ we likely need to call `d.sender.Commit()` to be sure that the metrics will be commited
2/ we might need a lock/mutex to prevent go routine accumulation just in case, ping doesn't timeout/return for a very long time. Rethinking about this, I'm not sure if it's needed, since ping command have a timeout, and will likely always return at worst around that timeout duration.

Reply:
The ping does have an enforced timeout, but I think it's still worth considering your suggestion.

The advantage with this is that we would avoid adding time to check duration unless the ping takes longer than the rest of the SNMP check. The downside is extra resource usage, but unless there's a massive number of check runners I don't know if its a problem.